src/frontend/src/components/EditTripsModal.tsx

```diff
@@ -1,32 +1,198 @@
-import { ModalOverlay, Modal } from "components/base/Modal/Modal";
+import { useEffect, useState } from "react";
+import type { Key } from "react-aria-components";
 import { Dialog, Heading } from "react-aria-components";
-import { useDispatch } from "react-redux";
+import { useTranslation } from "react-i18next";
+
 import {
     selectIsEditTripsOpen,
     setIsEditTripsOpen,
-} from "src/reducers/modalsDisplay/modalsDisplaySlice";
-import { useAppSelector } from "src/store/store";
+} from "reducers/modalsDisplay/modalsDisplaySlice";
+import {
+    selectTrafficConditions,
+    setTrafficConditions,
+} from "reducers/networks/networksSlice";
+import type { TrafficType } from "reducers/networks/types";
+import type { UserProfileSliceState } from "reducers/userProfile/types";
+import { updateUserProfile } from "reducers/userProfile/userProfileThunk";
+import { selectUserProfile } from "reducers/userProfile/userSlice";
+import { useAppDispatch, useAppSelector } from "store/store";
+
+import Button from "components/base/Button/Button";
+import { Modal, ModalOverlay } from "components/base/Modal/Modal";
+import {
+    ToggleButton,
+    ToggleButtonGroup,
+} from "components/base/ToggleButton/ToggleButton";
+import { modalHeadingClassName } from "pages/Discover/ModalsWrapper";
+import { Accordion } from "./base/Accordion/Accordion";
+import { AccordionItem } from "./base/Accordion/AccordionItem";
+import CommuterRailCheckbox from "./CommuterRailCheckbox";
 import ModalCloseButton from "./ModalCloseButton";
+import TravelModeToggle from "./TravelModeToggle";
 
 const EditTripsModal = () => {
-    const dispatch = useDispatch();
+    const { t } = useTranslation();
+    const dispatch = useAppDispatch();
     const modalOpen = useAppSelector(selectIsEditTripsOpen);
+    const profile = useAppSelector(selectUserProfile);
+    const trafficConditions = useAppSelector(selectTrafficConditions);
+    const [profileBuffer, setProfileBuffer] =
+        useState<UserProfileSliceState | null>(null);
```

Comment:
For the buffer implementations in the recent PRs, I wonder, why not initializing the `profileBuffer` by `profile` directly. Did you run into performance issues or other state management issues?